mautic/mautic
# Addon to plugin conversion

**Descriptions**
Addons are being officially renamed to plugins.  This PR makes the conversion. 

1) All instances of addon have been changed to plugin.
2) /addons have been relocated to /plugins
3) All plugins are enabled by default thus removed all code related to enabled/disabled plugin bundles
4) BC/deprecated support added for addons that leverage the MauticAddon prefix, extends the AddonBundleBase class, or uses the addon prefix for models, permissions, ajax, etc.  
5) Disabled lead profile for Facebook until the API restrictions are worked around
6) Added support for plugin schema installation or update through new inheritable onPluginInstall and onPluginUpdate functions in PluginBundleBase that are executed automatically when the reload button is clicked.  Doctrine metadata is used to install or update the plugin's schema.  
7) Removed the manage addon list view and made the integration view default.  Integrations that are disabled are grayscale; enabled/published are in color.  Plugins without integrations show in color and display a simple information modal when clicked rather than a form.

**Testing**
1) Backup the database as this PR includes a migration that is irreversible.
2) Apply the PR 
3) Run `composer install` as some of the autoload stuff has to be updated
and clear browser and symfony cache.
4) Run `php app/console doctrine:migrations:migrate` to update the schema

Instead of Addons in the admin menu, you should now see Plugins. Click to configure an integration or two and they should be in color.  Test the integrations such as Twitter lead profile to ensure all works OK.

Use https://github.com/drmmr763/TestAddonBundle to test BC compatibility.  The only BC break is if the addon references the path '/addons' directly rather than using `__DIR__` or similar.
